yt/utilities/parallel_tools/dask_helper.py

```diff
@@ -0,0 +1,145 @@
+from dask import array as dask_array, compute as dask_compute
+from dask.distributed import Client
+
+from yt.utilities.logger import ytLogger as mylog
+
+# a couple of warning messages used in several places
+here_be_dragons = (
+    "You have chosen to attempt multi-threading. Many frontends are not"
+    " thread-safe and many yt functions are built for single-threading so "
+    "you will likely encounter bugs (including silent failures and hangs)."
+    "Good luck!"
+)
+
+thread_override = (
+    "multi-threading is prohibited by default. Supply allow_threads=True "
+    "to start_client() to override if you really want to (you probably "
+    "dont want to)."
+)
+
+
+class ClientContainer:
+    """
+    a container to help with managing dask clients for yt. Enforces
+    single-threading for new clients and allows us to easily track if a client
+    is open.
+
+    Intentionally does NOT spin up a dask client on instantiation so that
+    a user is responsible for starting their own client. With no client,
+    the wrapper functions in dask_helper assume one single-threaded processor.
+    """
+
+    def __init__(self):
```

Comment:
We may need to do something to do our best to make sure this is a singleton.  I'm not 100% sure that this will be an issue, but in some cases I believe it is conceivable that the contents of a module may be executed multiple times.  This information may be outdated; however, in the case that it *can* still happen, it's not completely impossible that this could be run twice and the top-level `dask_client` instance may be different depending on which reference is utitilized.